Makefile

```diff
@@ -1,38 +1,44 @@
 PYJ_GLOBALS='$$,chrome,Marka,alertify'
+OUTPUT_DIR=output
+SOURCE_DIR=src
+VPATH = $(SOURCE_DIR)
 
 %.js: %.pyj
 	rapydscript lint --globals $(PYJ_GLOBALS) $(PYJ_FLAGS) $<
-	rapydscript $(PYJ_FLAGS) $< --output $@
-	sed -e ':a;N;$$!ba;s/async;\n/async/g' -i $@
+	rapydscript $(PYJ_FLAGS) $< --output $(OUTPUT_DIR)/$@
+	sed -e ':a;N;$$!ba;s/async;\n/async/g' -i $(OUTPUT_DIR)/$@
 
+JS_FILES=background.js roll20.js roll20_script.js \
+	fvtt.js fvtt_script.js fvtt_test.js \
+	dndbeyond_character.js dndbeyond_monster.js \
+	dndbeyond_spell.js dndbeyond_encounter.js dndbeyond_items.js  \
+	dndbeyond_vehicle.js options.js  popup.js default_popup.js \
+        dndbeyond.js roll_renderer.js
+PYJ_DEPS=utils.pyj settings.pyj dndbeyond.pyj dndbeyond_dice.pyj constants.pyj roll_renderer.pyj
 
-JS_FILES=src/background.js src/roll20.js src/roll20_script.js \
-	src/fvtt.js src/fvtt_script.js \
-	src/dndbeyond_character.js src/dndbeyond_monster.js \
-	src/dndbeyond_spell.js src/dndbeyond_encounter.js src/dndbeyond_items.js  \
-	src/dndbeyond_vehicle.js src/options.js  src/popup.js src/default_popup.js
-PYJ_DEPS=src/utils.pyj src/settings.pyj src/dndbeyond.pyj src/dndbeyond_dice.pyj src/constants.pyj src/roll_renderer.pyj
+fvtt_script.js: PYJ_GLOBALS='$$,chrome,game,canvas,Roll,ChatMessage,ui,Hooks,Dialog,ENTITY_PERMISSIONS,CONFIG,CHAT_MESSAGE_TYPES,CONST,CONFIG,isNewerVersion'
+utils.pyj-cached: PYJ_GLOBALS='$$,chrome,browser,cloneInto,alertify'
+constants.pyj-cached: PYJ_FLAGS+=--noqa eol-semicolon
 
-src/fvtt_script.js: PYJ_GLOBALS='$$,chrome,game,canvas,Roll,ChatMessage,ui,Hooks,Dialog,ENTITY_PERMISSIONS,CONFIG,CHAT_MESSAGE_TYPES,CONST,CONFIG,isNewerVersion'
-src/utils.pyj-cached: PYJ_GLOBALS='$$,chrome,browser,cloneInto,alertify'
-src/constants.pyj-cached: PYJ_FLAGS+=--noqa eol-semicolon
+all: lint $(JS_FILES)

```

Comment:
this depends on the files themselves but I don't see where it checks them from src/ directory. I see you use VPATH (which I didn't know of), but from a quick read, it looks like it's a search path for when files aren't found. Does this work to make sure it only rebuilds the appropriate files if the source file is modified ?

Reply:
I will test this, but the way I understand it, VPATH just appends a root directory the Makefile search path.
So I would imagine that the Makefile path by default is just:
`PATH=./` and by adding my VPATH dir, it becomes:
`PATH=./;src/`
As I understand, Makefile targets are always filenames relative to the root search directories.